RTNeural/activation/activation_eigen.h

```diff
@@ -464,6 +464,418 @@ class PReLUActivationT
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    SoftplusActivation(std::initializer_list<int> sizes)
+        : SoftplusActivation(*sizes.begin())
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+
+        /** Apply Softplus: ln(1 + exp(x)) */
+        outVec = (1 + MathsProvider::exp(inVec.array())).log(); //log() awaiting mathsprovider
+
+        std::copy(outVec.data(), outVec.data() + Layer<T>::in_size, out);
+    }
+
+    Eigen::Matrix<T, Eigen::Dynamic, 1> inVec;
+    Eigen::Matrix<T, Eigen::Dynamic, 1> outVec;
+};
+
+/** Static implementation of a Softplus activation layer. */
+template <typename T, int size, typename MathsProvider = DefaultMathsProvider>
+class SoftplusActivationT
+{
+    using v_type = Eigen::Matrix<T, size, 1>;
+
+public:
+    static constexpr auto in_size = size;
+    static constexpr auto out_size = size;
+
+    SoftplusActivationT()
+        : outs(outs_internal)
+    {
+        outs = v_type::Zero();
+    }
+
+    std::string getName() const noexcept { return "softplus"; }
+
+    constexpr bool isActivation() const noexcept { return true; }
+
+    RTNEURAL_REALTIME void reset() { }
+
+    /** Forward propagation for Softplus activation. */
+    RTNEURAL_REALTIME inline void forward(const v_type& ins) noexcept
+    {
+        /** Apply Softplus: ln(1 + exp(x)) */
+        outs = (1 + MathsProvider::exp(ins.array()).log(); //log() waiting for mathsprovider
+    }
+
+    Eigen::Map<v_type, RTNeuralEigenAlignment> outs;
+
+private:
+    T outs_internal alignas(RTNEURAL_DEFAULT_ALIGNMENT)[out_size];
+};
+
+/** Dynamic implementation of a Mish activation layer. */
+template <typename T, typename MathsProvider = DefaultMathsProvider>
+class MishActivation : public Activation<T>
+{
+public:
+    /** Constructor for Mish activation layer depending on the softplus activation internally */
+    explicit MishActivation(int size)
+        : Activation<T>(size, {}, "mish"), softplus(size)
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    MishActivation(std::initializer_list<int> sizes)
+        : MishActivation(*sizes.begin())
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+        /** mish = tanh(softplus(x)) */
+        Eigen::Matrix<T, Eigen::Dynamic, 1> softplusOut(Layer<T>::in_size);
+        softplus.forward(input, softplusOut.data());
```

Comment:
This seems fine, although I wonder if it might be simpler to factor out the logic of softplus into it's own function (maybe in `common.h`), and then call that from the layers that need it? Then we wouldn't need to have an "internal" instance of the full softplus layer class.

Reply:
I wonder how many people need softplus() as an activation function. It might be the case where not many people use it. And if so that could warrant it being in the common.h file